server/fleet/microsoft_mdm.go

```diff
@@ -341,3 +341,20 @@ type SoapFault struct {
 	Reason              Reason   `xml:"s:reason"`
 	OriginalMessageType int      `xml:"-"`
 }
+
+// MicrosoftMDMAccessTokenPayload is the payload that gets encoded as JSON and
```

Comment:
I'm going to merge this functionality with the server-side counterpart here 

https://github.com/fleetdm/fleet/pull/12477/files#diff-48f92386a8e8669440f0ee54c96b1bc848aae1201f3536aa24674f97f36b9a19R552